modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/spring/BeanDefinitionFactory.java

```diff
@@ -93,7 +107,7 @@ public BeanDefinition resolveComponentRecursively(ComponentModel parentComponent
         {
             for (ComponentModel innerComponent : innerComponents)
             {
-                if (hasDefinition(innerComponent.getIdentifier()))
+                if (hasDefinition(innerComponent.getIdentifier(), of(innerComponent.getParent().getIdentifier())))
```

Comment:
No, when it gets to this point the parent can't be null. That would be the case when the <mule> identifier is parsed but is filtered and never gets to this point.